pkg/proxy/iptables/proxier.go

```diff
@@ -500,26 +456,38 @@ func buildServiceMap(allServices []api.Service, oldServiceMap proxyServiceMap) (
 						apiservice.BetaAnnotationHealthCheckNodePort)
 				} else {
 					info.healthCheckNodePort = int(p)
+				}
+			}
+
+			oldInfo, exists := oldServiceMap[serviceName]
+			equal := reflect.DeepEqual(info, oldInfo)
```

Comment:
I checked that before doing this, I don't think it's necessary as DeepEqual already checks for nil and exits early.  Basically doing exactly what I would do if I open-coded it here.

```
func DeepEqual(x, y interface{}) bool {

		if x == nil || y == nil {

			return x == y

		}
```